### ISSUE TITLE
Error occurs in Windsurf when registering tool with input schema

### ISSUE THREAD

**Issue by Karibash** [modelcontextprotocol/typescript-sdk#800]
**Describe the bug**  
When registering a tool with an input schema, an error occurs specifically in Windsurf. This issue was reported in a library I maintain, and similar problems have been observed in other projects as well.

**To Reproduce**  
1. Register a tool with an input schema:
```ts
server.tool(
  'fetch',
  'fetch web page',
  { url: z.string().describe('web page url') },
  async ({ url }) => {
    const response = await fetch(url);
    return {
        content: [{ type: 'text', text: await response.text() }],
    };
  },
);
```

**Expected behavior**  
The tool should register correctly without causing an error in Windsurf.

**Logs**  
Unfortunately, I do not have direct access to logs since I do not use Windsurf myself. The issue was reported by users of my library.

**Additional context**  
- This issue was initially reported in my library: [praha-inc/byethrow#91](https://github.com/praha-inc/byethrow/issues/91)  
- A similar issue has been reported in another library as well: [makenotion/notion-mcp-server#17](https://github.com/makenotion/notion-mcp-server/issues/17)  
- The issue does not occur in VSCode or IntelliJ, suggesting it may be specific to Windsurf.  
- Since I don't use Windsurf personally, I can't provide further technical details, but I wanted to report this in case it's an SDK-side compatibility issue.

**Comment by michaelcatalano17** [comment]
I've been messing around with ollama and mcphost and am also running into this issue when trying to use the MCP server with mcphost. More specifically, I'm getting 
```
Error: failed to create agent: failed to create agent: failed to load MCP tools: all MCP servers failed to load: server notionapi: conv mcp tool input schema fail(unmarshal): json: cannot unmarshal array into Go struct field SchemaBis.properties.items.properties.properties.items.properties.properties.type of type string, tool name: API-patch-block-children
```
Just adding for visibility in case it helps anyone's investigation

**Comment by Karibash** [comment]
Since this was an issue on the Windsurf side and appears to have been fixed in v1.12.2, I will close it.
https://github.com/praha-inc/byethrow/issues/91#issuecomment-3216137672
